src/ast_traverse.ml

```diff
@@ -69,6 +69,44 @@ class map_with_path = object
     super#module_type_declaration (enter mtd.pmtd_name.txt path) mtd
 end
 
+let var_names_of = object
+  inherit [string list] fold as super
+
+  method! pattern p acc =
+    let acc = super#pattern p acc in
+    match p.ppat_desc with
+    | Ppat_var {txt; _} -> txt :: acc
+    | _ -> acc
+end
+
+class map_with_code_path = object (self)
+  inherit [Code_path.t] map_with_context as super
+
+  method! module_binding path mb =
+       super#module_binding (Code_path.enter mb.pmb_name.txt path) mb
+
+  method! module_declaration path md =
+    super#module_declaration (Code_path.enter md.pmd_name.txt path) md
+
+  method! module_type_declaration path mtd =
+    super#module_type_declaration (Code_path.enter mtd.pmtd_name.txt path) mtd
+
+  method! value_description path vd =
+    super#value_description (Code_path.enter vd.pval_name.txt path) vd
+
+  method! value_binding path vb =
```

Comment:
We should deconstruct `vb` in the pattern and without using `; _`. This way we are sure to process all the fields.

Reply:
Indeed that's much cleaner, I changed it!